tensorly/contrib/decomposition/tt_TTOI.py

```diff
@@ -0,0 +1,173 @@
+
+import numpy as np
+from math import prod
+import tensorly as tl
+from tensorly.tt_tensor import tt_to_tensor, validate_tt_rank
+import warnings
+
+
+#Author: Lili Zheng
+
+#Reference paper: Zhou Y, Zhang AR, Zheng L, Wang Y. "Optimal high-order tensor svd via tensor-train orthogonal iteration."
+def sequential_prod_reshape(X, M_arr, direction):
```

Comment:
Why not use n-mode product or `tensordot`?

Reply:
The `sequential_prod_reshape` function is different from n-mode product since the order of the multiplication matters. Although we can apply the `tensordot` function, we still need to write a separate function that calls `tensordot` sequentially, which is basically the same as what we have now.